File: 백준/Gold/13144. List of Unique Numbers/List of Unique Numbers.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(final String[] args) throws IOException {

        final BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        final int n = Integer.parseInt(st.nextToken());
        st = new StringTokenizer(br.readLine());

        final int[] nums = new int[n];
        final boolean[] used = new boolean[100_001];
        
        for (int i = 0; i < n; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }

        int one = 0;
        int two = 1;
        long res = 1;
        used[nums[one]] = true;
        while (two < n) {
            if (!used[nums[two]]) {
                used[nums[two]] = true;
                two += 1;                
                res += two - one;
            } else {
                used[nums[one]] = false;
                one += 1;
            }
        }
        System.out.print(res);
    }
}

File: 백준/Gold/13144. List of Unique Numbers/List of Unique Numbers.py
import sys
input = sys.stdin.readline

n = int(input())
nums = list(map(int,input().split()))

one = 0
two = 1
ss = {nums[one]}
res = 1
while two < n:
    if nums[two] not in ss:
        ss.add(nums[two])
        two += 1
        res += two - one
    else:
        ss.remove(nums[one])
        one += 1
print(res)
